src/arrayMethodJoin.js

```diff
@@ -6,6 +6,24 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    let joinedStr = '';
+    const arrLength = this.length;
+
+    for (let i = 0; i < arrLength; i++) {
+      const type = typeof (this[i]);
+
+      if (type === 'number' || type === 'string' || type === 'boolean') {
+        joinedStr += this[i];
+      }
+
```

Comment:
I think it will be enough to check if `this[i]` is `null `or `undefined`, if so skip the iteration to the next using `continue`